tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_SchwarzAlgorithm.cpp

```diff
@@ -40,9 +40,92 @@ struct SchwarzSmoother {
       "Options for the iterative Schwarz smoother";
 };
 
+template <typename DirectionsTag>
+struct DoNothing {
+  using argument_tags = tmpl::list<>;
+  template <typename SubdomainData, typename ResultData,
+            typename SubdomainOperator>
+  static void apply(
+      const SubdomainData& /*subdomain_data*/,
+      const gsl::not_null<ResultData*> /*result*/,
+      const gsl::not_null<SubdomainOperator*> /*subdomain_operator*/) noexcept {
+  }
+};
+
+DenseMatrix<double> combine_matrix_slices(
+    const std::vector<DenseMatrix<double>>& matrix_slices) {
+  const size_t num_slices = matrix_slices.size();
+  const size_t num_cols_per_slice = matrix_slices.begin()->columns();
+  const size_t total_num_points = num_slices * num_cols_per_slice;
+  DenseMatrix<double> full_matrix(total_num_points, total_num_points);
+  for (size_t i = 0; i < num_slices; ++i) {
+    blaze::submatrix(full_matrix, 0, i * num_cols_per_slice, total_num_points,
+                     num_cols_per_slice) = gsl::at(matrix_slices, i);
+  }
+  return full_matrix;
+}
+
+template <typename Tag>
+DenseVector<double> extend_subdomain_data(
+    const LinearSolver::Schwarz::ElementCenteredSubdomainData<
+        1, tmpl::list<Tag>>& subdomain_data,
+    const size_t element_index, const size_t num_elements,
+    const size_t num_points_per_element, const size_t overlap_extent) {
+  DenseVector<double> extended_subdomain_data(
```

Comment:
Can you use spectre vector types for this? Maybe with the pointer-and-size constructor to create a referencing 'view'?

Reply:
Maybe, though that would replace the `blaze::subvector` calls with our own nonowning-vector functionality. Since this test works entirely with matrices and vectors, I would just stick with the standard blaze matrix and vector types (`DenseVector` and `DenseMatrix` are just thin wrappers that enable things like make_with_value and option-creation).